### PR TITLE
core: support for request IDs in non-error streaming case

### PR DESCRIPTION
**Pull request by spolu**
## Description

Leverage new `SSE::Connected` event to pull the header from providers who provide request IDs in headers (openai, anthropic, mistral).

## Risk

Low, tested locally

## Deploy Plan

- deploy `core`

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -52,7 +52,7 @@ urlencoding = "2.1"
 url = "2.5"
 dns-lookup = "1.0"
 async-stream = "0.3"
-eventsource-client = { git = "https://github.com/dust-tt/rust-eventsource-client", rev = "d34896e77b2198cd4d71e05007798736142ec336" }
+eventsource-client = { git = "https://github.com/dust-tt/rust-eventsource-client", rev = "148050f8fb9f8abb25ca171aa68ea817277ca4f6" }
```
Comment (fontanierh) [inline core/Cargo.toml:55]: Gonna have to fight the LSP again :p

File: core/src/providers/anthropic.rs [hunk 2]
```diff
@@ -805,10 +805,18 @@ impl AnthropicLLM {
...
 
         let mut final_response: Option<StreamChatResponse> = None;
+        let mut request_id: Option<String> = None;
+
         'stream: loop {
             match stream.try_next().await {
                 Ok(stream_next) => {
                     match stream_next {
+                        Some(es::SSE::Connected((_, headers))) => {
+                            request_id = match headers.get("request-id") {
```
Comment (fontanierh) [inline core/src/providers/anthropic.rs:815]: nit: could use map

File: core/src/providers/anthropic.rs [hunk 3]
```diff
@@ -1148,9 +1156,17 @@ impl AnthropicLLM {
 
         let mut final_response: Option<CompletionResponse> = None;
         let mut completion = String::new();
+        let mut request_id: Option<String> = None;
+
         'stream: loop {
             match stream.try_next().await {
                 Ok(stream_next) => match stream_next {
+                    Some(es::SSE::Connected((_, headers))) => {
+                        request_id = match headers.get("request-id") {
```
Comment (fontanierh) [inline core/src/providers/anthropic.rs:1165]: nit map

**Review by fontanierh — APPROVED** [review]
LGTM

File: core/Cargo.toml (hunk 1 above)
Comment (spolu) [reply core/Cargo.toml:55]: This time for good!

Comment (spolu) [reply core/Cargo.toml:55]: (moved from commit to rev which is what we wanted all along)

**Pull request merged**